app/src/main/python/gogdl/xdelta/patcher.py

```
@@ -0,0 +1,204 @@
+            context.source.seek(context.cpy_off + halfinst.addr)
+            left = take
+            while left > 0:
+                buffer = context.source.read(min(1024 * 1024, left))
+                size = len(buffer)
+                speed_queue.put((0, size))
+                context.target_buffer.extend(buffer)
+                left -= size
+
+        else:
+            print("OVERLAP NOT IMPLEMENTED")
+            raise Exception("OVERLAP")
```

Ruff (0.12.2)

Create your own exception

(TRY002)